addons/storyshots/src/test-body-image-snapshot.js

```diff
@@ -0,0 +1,66 @@
+import puppeteer from 'puppeteer';
+import { toMatchImageSnapshot } from 'jest-image-snapshot';
+
+expect.extend({ toMatchImageSnapshot });
+
+export const imageSnapshot = ({ port = 6006, host = 'localhost', scheme = 'http' }) => {
+  let browser; // holds ref to browser. (ie. Chrome)
+  let page; // Hold ref to the page to screenshot.
+
+  const testFn = ({ context }) => {
+    if (context.isRNStorybook) {
+      // Skip tests since we de not support RN image snapshots.
+      console.error(
+        "It seems you are running imageSnapshot on RN app and it's not supported. Skipping test."
+      );
+      return Promise.resolve();
+    }
+    const baseUrl = `${scheme}://${host}:${port}`;
+    const encodedKind = encodeURIComponent(context.kind);
+    const encodedStoryName = encodeURIComponent(context.story);
+    const storyUrl = `/iframe.html?selectedKind=${encodedKind}&selectedStory=${encodedStoryName}`;
+    const url = baseUrl + storyUrl;
+    if (!browser || !page) {
+      console.error(
+        `Error when generating image snapshot for test ${context.kind} - ${
+          context.story
+        } : It seems the headless browser is not running.`
+      );
+      return Promise.reject(new Error('no-headless-browser-running'));
+    }
+
+    expect.assertions(1);
+    return page
+      .goto(url)
+      .catch(e => {
+        console.error(
+          `ERROR WHILE CONNECTING TO ${
+            url
+          }, did you start the storybook first ? A storybook instance should be running when using image snapshot feature.`,
+          e
+        );
+        throw e;
+      })
+      .then(() =>
+        page.screenshot().then(image => {
+          expect(image).toMatchImageSnapshot();
```

Comment:
This really needs to be configurable (see https://github.com/americanexpress/jest-image-snapshot#optional-configuration)

Reply:
The main issue is what @tobilen brought up: text renders differently on different OS. So setting `failureThreshold` would be needed most of the times